practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/service/PractitionerService.java

```diff
@@ -75,6 +83,21 @@ public Practitioner findPractitionerById(String id) throws ResponseStatusExcepti
 
     }
 
+    @Override
+    public void linkPractitioner(PractitionerIdProviderLink link) {
+        ValidationResponse validationResponse = directoryLinkValidator.validate(link);
+
+        if (!validationResponse.isValid()) {
+            throw new ValidationException(validationResponse.getErrorMessage());
+        }
+        
+        final Practitioner practitioner = practitionerStore.getEntity(link.getPractitionerId()).orElseThrow(() ->
+            new ResponseStatusException(HttpStatus.NOT_FOUND, Services.PRACTITIONER_NOT_FOUND.message()));
```

Comment:
Not sure if it's better or not, but when I needed the get a practitioner in saveRoleAssignment, I just called the method in this class instead of going through the store.